### PR TITLE
Frontend logging fixes

### PR DESCRIPTION
**Pull request by achen2401**
Address https://www.pivotaltracker.com/story/show/188705822
- Add user info to each event logging
- Add `analytics` tag to tab/open/view events
- Prevent `cosri-frontend` tag from being added multiple times

Example JSONs posted

  ```
{
    "patient": "Barney Abbott",
    "message": "MME calculated: Name: Fentanyl Transdermal System, 0.05 mg/hr...",
    "level": "info",
    "tags": ["mme-calc", "cosri-frontend"],
    "patientName": "Barney Abbott",
    "subject": "Patient/5ee05359-57bf-4cee-8e89-91382c07e162",
    "user": { "username": "amysbubble" }
  }
```
  ```
{
    "patient": "Barney Abbott",
    "message": "overview tab",
    "level": "info",
    "tags": ["tab", "analytics", "cosri-frontend"],
    "patientName": "Barney Abbott",
    "subject": "Patient/5ee05359-57bf-4cee-8e89-91382c07e162",
    "user": { "username": "amysbubble" }
  }
```

### CODE REVIEW

**Review by mcjustin — APPROVED** [review]

File: src/helpers/utility.js [hunk 1]
```diff
@@ -573,10 +573,18 @@ export function writeToLog(message, level, params) {
...
-  logParams.tags.push("cosri-frontend");
-
+  const COSRI_FRONTEND_TAG = "cosri-frontend";
+  if (logParams.tags.indexOf(COSRI_FRONTEND_TAG) === -1) {
+    logParams.tags.push(COSRI_FRONTEND_TAG);
+  }
+  if (Object.keys(logParams).indexOf("user") === -1) {
+    const userId = getUserIdFromAccessToken();
+    if (userId)
+      logParams.user = userId;
```
Comment (ivan-c) [inline src/helpers/utility.js:586]: Could to nest the username under `user`? 

I realized we're already following this pattern for cosri-patientsearch and the SoF backend

```suggestion
      logParams.user.username = userId;
```

**Review by ivan-c — CHANGES_REQUESTED** [review]
Thank you! I tested on dev and found a minor inconsistency with the rest of our logs

**Comment by achen2401** [comment]
thank you @ivan-c for catching the inconsistency. I updated the description with updated user format in the JSON, e.g.
```
{
....
"user": { "username": "amysbubble" }
}
```

File: src/helpers/utility.js [hunk 2]
```diff
@@ -573,10 +573,21 @@ export function writeToLog(message, level, params) {
...
+  const logParams = Object.assign({}, params ? params : {});
   if (!logParams.tags) logParams.tags = [];
-  logParams.tags.push("cosri-frontend");
-
+  const COSRI_FRONTEND_TAG = "cosri-frontend";
+  if (logParams.tags.indexOf(COSRI_FRONTEND_TAG) === -1) {
+    logParams.tags.push(COSRI_FRONTEND_TAG);
+  }
+  if (Object.keys(logParams).indexOf("user") === -1) {
+    const userId = "amysbubble";
```
Comment (ivan-c) [inline src/helpers/utility.js:584]: I think this was leftover from testing
```suggestion
    const userId = getUserIdFromAccessToken();
```

**Review by ivan-c — APPROVED** [review]
Found a minor issue. Test on dev and looks good otherwise!

**Pull request merged**